JavaCryptographicArchitecture/src/KeyFactory.crysl

```diff
@@ -0,0 +1,28 @@
+SPEC java.security.KeyFactory
+OBJECTS
+    java.lang.String keyFactoryAlgorithm;
+    java.security.spec.KeySpec keySpec;
+    java.security.PrivateKey privKey;
+    java.security.PublicKey pubKey;
+   
+EVENTS
+    g1: getInstance(keyFactoryAlgorithm);
+    g2: getInstance(keyFactoryAlgorithm, _);
+    Gets := g1 | g2;
+
+    gPriv: privKey = generatePrivate(keySpec);
+    gPubl: pubKey = generatePublic(keySpec);
+    
+ORDER
+    Gets, (gPriv* | gPubl*)*
+
+CONSTRAINTS
+	keyFactoryAlgorithm in {"RSA", "DiffieHellman", "DSA", "EC"};

```

Comment:
Does DH work as a synonym here as well?

Reply:
yes it works. Commit: https://github.com/CROSSINGTUD/Crypto-API-Rules/commit/e1fc6ce4d73609a8d96cf717534c65836f616467